src/prototype.rs

```diff
@@ -104,7 +104,7 @@ enum ItemBody {
     Strong,
     Rule,
     Header(i32), // header level
-    FencedCodeBlock(usize, u8, usize), // number of fence chars, fence char, indentation
+    FencedCodeBlock(usize, u8, usize, String), // number of fence chars, fence char, indentation, info string
```

Comment:
The quick answer to the lifetime question is that we _could_ make the enum `ItemBody<'a>` where `'a` is the lifetime of the source text, and make this a `Cow<'a, str>` instead. However, I'm not sure the performance gain is worth the extra hassle. I'm happy leaving it as is, and it's a potential optimization for later.